src/UhdmAst.cpp

```diff
@@ -247,6 +247,32 @@ AstNode* get_value_as_node(vpiHandle obj_h, bool need_decompile = false) {
     return valueNodep;
 }
 
+AstNodeDType* get_packed_array_dtype(vpiHandle obj_h, UhdmShared& shared, AstBasicDType* basic) {
```

Comment:
Is a separate function necessary? The only difference is how to get the initial type, once that is done we can handle that in one case in `getDType`.

Reply:
I created a separate function, because I didn't want to add `if`s for the types inside one case. I thought that if a few types are handled in one case, they should be handled identically. 
If you think that it would be better to create `if` for `vpiPackedArrayTypespec`, I can do it.